src/snakeoil/mappings.py

```diff
@@ -41,7 +42,7 @@ class DictMixin:
     `values` and `items` in addition.
     """
 
-    __slots__ = ()
+    __slots__ = ("__weakref__",)
```

Comment:
having come back out of the rabit hole, in hindsight I think this should be a choice of the childclass, so I'll back this out.